### ISSUE TITLE
mAP% at zero all time

### ISSUE THREAD

**Issue by lucaspojo** [AlexeyAB/darknet#5436]
Hi ! 
I've been using yolo for a little while now and I have a new problem. I'm trying to **detect license plates** in 1920x1080 pictures.
I created a dataset with LabelImg, as usual, I configured the .CFG as usual with a few modifications. And here's the result :

![mAP](https://zupimages.net/up/20/18/u8mm.png)

CFG for training : https://gist.github.com/lucaspojo/ac991b43501587eaac81426c82e790b3

Sample of my dataset : https://drive.google.com/drive/folders/1oUj7y1DPfJ9K6UVKJU4kiPGB7uHCuSbq?usp=sharing

In the end, the detection still works! So I don't understand where the trouble is. I would like to have a functional mAP. Thanks for your help!

Youtube demo : https://www.youtube.com/watch?v=j9S-qudNONs

ps : Is use Amazon GPU for this training or my own RTX 2060. I have 1400 Labeled images and 200 for testing.

**Comment by AlexeyAB** [comment]
Check the mAP by using `./darknet detector map ...` command, if the mAP is 0, then there is wrong validation dataset or wrong path in `obj.data` file

**Comment by lucaspojo** [comment]
Output : 
>  calculation mAP (mean average precision)...
> 800
>  detections_count = 455917, unique_truth_count = 1020  
> class_id = 0, name = imat, ap = 0.00%   	 (TP = 0, FP = 455917) 
> 
>  for conf_thresh = 0.25, precision = 0.00, recall = 0.00, F1-score = -nan 
>  for conf_thresh = 0.25, TP = 0, FP = 455917, FN = 1020, average IoU = 0.00 % 
> 
>  IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
>  mean average precision (mAP@0.50) = 0.000000, or 0.00 % 
> Total Detection Time: 109 Seconds
> 
> Set -points flag:
>  `-points 101` for MS COCO 
>  `-points 11` for PascalVOC 2007 (uncomment `difficult` in voc.data) 
>  `-points 0` (AUC) for ImageNet, PascalVOC 2010-2012, your custom dataset
>

**Comment by AlexeyAB** [comment]
Your validation dataset is wrong or wrong path in obj.data file

**Comment by lucaspojo** [comment]
It's done, i juste forgot to change the directory in .data file. Working great now. **Thx again and again**.

**Comment by ToCHONG** [comment]
> It's done, i juste forgot to change the directory in .data file. Working great now. **Thx again and again**.

Hi @lucaspojo,
I've come across the same issue, and I checked my validation dataset and path too. But the problem still persist and sometimes the calculation doesn't even finish without  outputting errors. Could you please help me with that?
Here's my .data file:
` 
  classes= 24
  train  = c:\test\train.txt
  valid  = c:\Users\z5539\Desktop\test.txt
  names = c:\Users\z5539\Desktop\darknet\build\darknet\x64\cfg\ts.names
  backup = backup
`

Content in the test.txt file is like the following:
`
c:/Users/z5539/Desktop/test/10056.jpg
........
c:/Users/z5539/Desktop/test/10063.jpg
`
And I have my dataset as follows:
![image](https://user-images.githubusercontent.com/41679641/81265132-1ff5d780-9075-11ea-988d-3bbf8a593cce.png)

At last, my calculation command:
`darknet.exe detector map cfg/ts.data cfg/yolov3-ts.cfg backup/yolov3-ts_final.weights -points 0`
Thanks for helping me!

**Comment by lucaspojo** [comment]
Can us show me the content of **ts.data** ?

**Comment by lucaspojo** [comment]
@ToCHONG Can you show me the content of **ts.data** ?

**Comment by ToCHONG** [comment]
Thanks for the quick reply @lucaspojo.
I've already get the final weights which seems doing fine in other images. So the training process is not the problem. The problem is only the calculation of mAP. And I still got this after adding the -map flag(same as before):
![image](https://user-images.githubusercontent.com/41679641/81277268-c1395980-9086-11ea-973c-98f9ca55cf28.png)
The sample of the validation dataset labels is as follws:
![image](https://user-images.githubusercontent.com/41679641/81277391-ef1e9e00-9086-11ea-9577-12b1293fe7b0.png)
I think maybe the problem is the number of digits? But the training set label is the same situation and works well. @AlexeyAB Is that a bug or it's just my mistake? Thanks again!

**Comment by ToCHONG** [comment]
Thanks for your quick reply. Problem solved and it's my mistake haha. Thanks anyway!

**Comment by frankissimo** [comment]
I have the same problem, mAP always at 0, labels in the valid test generated correctly (they are generated in the same way as the training set) and path in data.obj correct (I tried to delete the file and it gives me an error as it cannot be found). I tried to do a test and I saw that the bounding boxes are correct even in the case of predictions. I don't know what to feel anymore = (

**Comment by dashos18** [comment]
> I have the same problem, mAP always at 0, labels in the valid test generated correctly (they are generated in the same way as the training set) and path in data.obj correct (I tried to delete the file and it gives me an error as it cannot be found).
> I try the command detector map and if I test only half of the validation set, it works, if I test it all it doesn't work !

The same for me! Did you solve it?
